test/functionalTest/cases/2015_notification_templates/notification_templates_error_handling.test

```diff
@@ -0,0 +1,274 @@
+echo "================================================================================"
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://127.0.0.1:'${LISTENER_PORT}'/${NOTIFY_URL}"
+    },
+    "httpExtended": {
+      "url": "http://127.0.0.1:'${LISTENER_PORT}'/${NOTIFY_URL}"
+    }
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+--REGEXPECT--
+01. Create subscription for E1 with empty template part - see error
+===================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 79
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "url httpExtended notification is missing",
+    "error": "BadRequest"
+}
+
+
+02. Create subscription for E1 with everything except URL in template part - see error
+======================================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 79
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "url httpExtended notification is missing",
+    "error": "BadRequest"
+}
+
+
+03. Create subscription for E1 with URL and invalid VERB in template part - see error
+=====================================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 79
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "unknown method httpExtended notification",
+    "error": "BadRequest"
+}
+
+
+04. Create subscription for E1 without URL in http-part - see error
+===================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 71
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "url http notification is missing",
+    "error": "BadRequest"
+}
+
+
+05. Create subscription for E1 without http-part nor template part - see error
+==============================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 67
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "http notification is missing",
+    "error": "BadRequest"
+}
+
+
+06. Create subscription for E1 with both http-part and template part - see error
+================================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 77
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "notification has http and httpExtended",
+    "error": "BadRequest"
+}
+
+
+--TEARDOWN--
+brokerStop CB
+dbDrop CB
```

Comment:
I miss several test cases:

* N initial notif with substitution
* N initial notif without substitution
* Get subscriptions to check that httpExtended is properly rendered
* Text with a XML-based payload
* Text with a JSON-baesd payload

(You also mentioned some test at skype related with cache and refresh... they should be probably added to the list).

I understand that this PR is not the place to complete them, but maybe it can be used to create the "stub" (in .DISABLED way) including only the header and steps sequence for each one so we can discuss that sequence. Next PR will be focused on de-.DISABLED these test.